aws/src/main/java/org/apache/iceberg/aws/s3/signer/S3V4RestSignerClient.java

```diff
@@ -81,13 +77,14 @@ public abstract class S3V4RestSignerClient
   private static final String SCOPE = "sign";
 
   @SuppressWarnings("immutables:incompat")
-  private static volatile ScheduledExecutorService tokenRefreshExecutor;
+  private volatile AuthManager authManager;
```

Comment:
related to my other comment from https://github.com/apache/iceberg/pull/12197/files#r1993040250 I currently don't know whether having this field being non-static would cause what was earlier tested in https://github.com/apache/iceberg/pull/12197/files#r1993040250